api/envoy/config/core/v3/substitution_format_string.proto

```diff
@@ -61,6 +64,12 @@ message SubstitutionFormatString {
     //  }
     //
     google.protobuf.Struct json_format = 2 [(validate.rules).message = {required: true}];
+
+    // A DataSource variant for text_format.
+    DataSource text_format_source = 5;
+
+    // The empty format string.
+    bool empty_format = 6 [(validate.rules).bool = {const: true}];
```

Comment:
I don't see where this is used, shall we just relax the min_len: 1 constraint on text_format if we want an empty format string?

Reply:
We use it here https://github.com/envoyproxy/envoy/pull/14221/files#diff-7a3f06d418ccdf0de0f51a473b6d3450baa5faf9ac8362aa6fe33bc1284b5152R27

We could just relax the constraint, but my initial idea was to avoid making changes to an existing field and figured the explicit "empty" option was clear enough.

Reply:
I would slightly prefer relax the existing one, or we can have a field `fixed_text` which just do what PlainStringFormatterImpl does.